packages/common/core-utils/src/assert.ts

```diff
@@ -110,21 +123,22 @@ export function onAssertionFailure(handler: (error: Error) => void): () => void
 /**
  * Asserts that can be conditionally enabled in debug/development builds but will be optimized out of production builds.
  *
- * Disabled by default.
+ * Enabled when {@link nonProductionConditionalsIncluded} is true.
  *
  * If the assert must be enforced/checked in production or enabled by default, use {@link assert} instead.
  *
  * @param predicate - A pure function that should return true if the condition holds, or a string or object describing the condition that failed.
  * This function will only be run in some configurations so it should be pure, and only used to detect bugs (when debugAssert are enabled), and must not be relied on to enforce the condition is true: for that use {@link assert}.
  * @remarks
- * Optimizing the asserts out of the bundle requires a bundler like webpack which leverages `__PURE__` annotations like https://webpack.js.org/guides/tree-shaking/#mark-a-function-call-as-side-effect-free.
- *
  * Exceptions thrown by this function must never be caught in production code, as that will result in different behavior when testing and when running optimized builds.
  * The `predicate` function must be pure (have no side-effects) to ensure that the behavior of code is the same regardless of if the asserts are disabled, enabled or optimized out.
  *
- * These asserts are disabled by default, even in debug builds to ensure that by default code will be tested as production runs, with them disabled.
- * Additionally, this ensures that apps that use a bundler which does not remove `__PURE__` will not incur the runtime cost of calling the predicate.
- * These asserts can be can be enabled by calling `configureDebugAsserts(true)`: see {@link configureDebugAsserts}.
+ * These asserts are enabled by default in debug builds: this introduces risk that code may behave differently when they are disabled or optimized out.
+ * To mitigate this risk, these asserts can be disabled in debug builds by calling {@link configureDebugAsserts} or {@link emulateProductionBuild}.
+ * This allows testing with the asserts both enabled and disabled to help ensure that code does not depend on them being enabled.
+ *
+ * Apps (or other performance sensitive scenarios) packaged in a way that does not {@link nonProductionConditionalsIncluded|non-production code}
```

Comment:
I think this is right?
```suggestion
 * Apps (or other performance sensitive scenarios) packaged in a way that does not {@link nonProductionConditionalsIncluded|skip non-production code}
```